src/core/importers/specs.ts

```diff
@@ -34,12 +35,19 @@ export const importSpecs = async (
     } else {
       try {
         const { headers, data: specification } = await axios.get(path);
+        const isContentTypeYaml = headers['content-type'].includes('text/yaml');
+        const isContentTypeTextPlain = headers['content-type'].includes(
+          'text/plain',
+        );
+        const isContentTypeJson = headers['content-type'].includes(
+          'application/json',
+        );
 
-        if (
-          headers['content-type'] === 'application/json' ||
-          headers['content-type'].includes('text/plain')
-        ) {
+        if (isContentTypeJson || isContentTypeYaml || isContentTypeTextPlain) {
           data = specification;
+          if (isContentTypeTextPlain && typeof data === 'string') {
```

Comment:
you need to do that for isContentTypeJson and isContentTypeYaml too. Just assign the format directly (format = 'json' for isContentTypeJson).